src/Commands/AutoLoad.cpp

```diff
@@ -0,0 +1,270 @@
+}
+
+const wchar_t *AutoLoadCommandClass::GetUICategory() const
+{
+	return CATEGORY_SELECTION;
+}
+
+const wchar_t *AutoLoadCommandClass::GetUIDescription() const
+{
+	return GeneralUtils::LoadStringUnlessMissing("TXT_AUTO_LOAD_DESC", L"Auto Load");
+}
+
+void DebugPrintTransport(std::vector<std::pair<TechnoClass *, int>> &tTransports)
+{
+	Debug::Log("AutoLoadCommandClass::DebugPrintTransport: Transport count: %d\n", tTransports.size());
+	// print address of each transport and its passengers
+	for (auto transport : tTransports)
+	{
+		Debug::Log("AutoLoadCommandClass::DebugPrintTransport: Transport address: %p, Now size: %d, Virtual size: %d\n", transport.first, transport.first->Passengers.GetTotalSize(), transport.second);
+	}
+}
+
+void DebugPrintPassenger(std::vector<TechnoClass *> &pPassengers)
+{
+	Debug::Log("AutoLoadCommandClass::DebugPrintPassenger: Passenger count: %d\n", pPassengers.size());
+	// print address of each passenger
+	for (auto passenger : pPassengers)
+	{
+		Debug::Log("AutoLoadCommandClass::DebugPrintPassenger: Passenger address: %p\n", passenger);
+	}
+}
+
+template <typename P, typename T>
+void SpreadPassengersToTransports(std::vector<P> &pPassengers, std::vector<std::pair<T, int>> &tTransports)
+{
+	// 1. Get the least kind of passengers
+	// 2. Send the passengers to the transport in round robin, if the transport is full, remove it from the vector tTransports;
+	// if the pID vector is empty, remove it from the map, if not, move it to passengerMapIdle. We try to fill the transport evenly for each kind of passengers.
+	// 3. Repeat until all passengers are sent or the vector tTransports is empty.
+	std::unordered_map<const char *, std::vector<P>> passengerMap;
+	std::unordered_map<const char *, std::vector<P>> passengerMapIdle;
+
+	for (auto pPassenger : pPassengers)
+	{
+		auto pID = pPassenger->get_ID();
+		if (passengerMap.find(pID) == passengerMap.end())
+		{
+			passengerMap[pID] = std::vector<P>();
+		}
+		passengerMap[pID].push_back(pPassenger);
+	}
+
+	while (true)
+	{
+		while (passengerMap.size() > 0 && tTransports.size() > 0)
+		{
+			const char *leastpID = nullptr;
+			unsigned int leastSize = std::numeric_limits<unsigned int>::max();
+			for (auto const &[pID, pPassenger] : passengerMap)
+			{
+				if (pPassenger.size() < leastSize)
+				{
+					leastSize = pPassenger.size();
+					leastpID = pID;
+				}
+			}
+
+			{
+				unsigned int index = 0;
+				for (; index < leastSize && index < tTransports.size(); index++)
+				{
+					auto pPassenger = passengerMap[leastpID][index];
+					auto pTransport = tTransports[index].first;
+					auto pTypeExt = TechnoTypeExt::ExtMap.Find(pTransport->GetTechnoType());
+					bool bySize = true;
+					// Check passenger filter here.
+					if (pTypeExt)
+					{
+						if (!pTypeExt->CanLoadPassenger(pTransport, pPassenger))
+						{
+							continue;
+						}
+						bySize = pTypeExt->Passengers_BySize;
```

Comment:
```suggestion
						if (!pTypeExt->CanLoadPassenger(pTransport, pPassenger))
							continue;

						bySize = pTypeExt->Passengers_BySize;
```

Please check the style guide on this.